src/Output.hs

```diff
@@ -552,27 +534,60 @@ findFstFreeSlot table n = do
 -- Convert an integer to a 16-bit number encoded in \xNN\xNN format suitable
 -- for placing in a string (copied from Happy's ProduceCode.lhs)
 
+-- | Lay out string literal consisting of hexadecimal characters into columns
+-- of specified width.
+concatInChunks :: Int -> [HexChar] -> String
+concatInChunks width =
+  -- Take care to not use split string syntax, e.g.
+  -- x = "foo\
+  --     \bar"
+  -- because it does not play well with the preprocessor, which is always
+  -- enabled in the generated file.
+  intercalate "\\\n" .
```

Comment:
I'd want to make sure that works with clang and gcc on Linux as well as with Apple's clang from XCode.